lib/Value/Matrix.pm

```diff
@@ -327,24 +336,31 @@ sub typeRef {
 
 =head3 C<isSquare>
 
-Return true is the matrix is 1D of length 1 or 2D and square, false otherwise
+Return true if the Matrix is degree 1 of length 1 or its last two dimensions are equal, false otherwise
 
 Usage:
 
-    $A = Matrix([ [ 1, 2, 3, 4 ], [ 5, 6, 7, 8 ], [ 9, 10, 11, 12 ] ]);
-    $B = Matrix([ [ 1, 0, 0 ], [ 0, 1, 0 ], [ 0, 0, 1 ] ]);
+    $A = Matrix([ [ 1, 2, 3 ], [ 4, 5, 6 ], [ 7, 8, 9 ] ]);
+    $B = Matrix([ [ 1, 0, 0 ], [ 0, 1, 0 ] ]);
+    $C = Matrix(1);
+    $D = Matrix(1, 2);
+    $E = Matrix([ [ [ 1, 2 ], [ 3, 4 ] ] ]);
+    $F = Matrix([ [ [ 1, 2 ] ], [ [ 3, 4 ] ] ]);
 
-    $A->isSquare; # is '' (false)
-    $B->isSquare; # is 1 (true);
+    $A->isSquare; # is 1  (true)  because it is a 3x3 matrix
+    $B->isSquare; # is '' (false) because it is a 2x3 matrix
+    $C->isSquare; # is 1  (true)  because it is a degree 1 matrix of length 1
+    $D->isSquare; # is '' (false) because it is a degree 1 matrix of length 2
+    $E->isSquare; # is 1  (true)  because it is a 1x2x2 tensor
+    $F->isSquare; # is '' (false) because it is a 2x1x2 tensor
 
 =cut
 
 sub isSquare {
 	my $self = shift;
 	my @d    = $self->dimensions;
-	return 1 if scalar(@d) == 1 && $d[0] == 1;
-	return 0 if scalar(@d) != 2;
-	return $d[0] == $d[1];
+	if (scalar(@d) == 1) { return $d[0] == 1 }
+	return ($d[-1] == $d[-2]);
```

Comment:
I just noticed this (while looking at #1216).  I would change this to
```suggestion
	return $d[0]  == 1 if scalar(@d) == 1;
	return $d[-1] == $d[-2];
```
The first line is consistent with how it was before and how conditionals like this are generally done in this file.
The parentheses are just unnecessary on the second line.